core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1313,11 +1314,31 @@ private void checkConsistencyRecursive(LockedInodePath inodePath,
       }
       if (inode.isDirectory()) {
         InodeDirectory inodeDir = inode.asDirectory();
-        for (Inode child : mInodeStore.getChildren(inodeDir)) {
+        Iterable<? extends Inode> children = mInodeStore.getChildren(inodeDir);
+        for (Inode child : children) {
           try (LockedInodePath childPath = inodePath.lockChild(child, LockPattern.READ)) {
             checkConsistencyRecursive(childPath, inconsistentUris, assertInconsistent);
           }
         }
+        // if a file exist in mount table but not exist in alluxio
+        // it should as regard as inconsistent.
+        // if it's a directory, we could ignore child path.
+        MountTable.Resolution resolution = mMountTable.resolve(inodePath.getUri());
+        try (CloseableResource<UnderFileSystem> ufsResource = resolution.acquireUfsResource()) {
+          UnderFileSystem ufs = ufsResource.get();
+          String ufsPath = resolution.getUri().getPath();
+          UfsStatus[] statuses = ufs.listStatus(ufsPath);
+          if (statuses != null) {
+            Arrays.stream(statuses).forEach(status -> {
```

Comment:
A little worried about the O(n^2) time complexity to compare the inconsistency status.

Reply:
Have you tested the functionality?  i am asking because i am not quite sure if 
```
  public List<AlluxioURI> checkConsistency(AlluxioURI path, CheckConsistencyContext context)
      throws AccessControlException, FileDoesNotExistException, InvalidPathException, IOException {
    List<AlluxioURI> inconsistentUris = new ArrayList<>();
    try (RpcContext rpcContext = createRpcContext(context);
        FileSystemMasterAuditContext auditContext =
            createAuditContext("checkConsistency", path, null, null)) {

      syncMetadata(rpcContext,
          path,
          context.getOptions().getCommonOptions(),
          DescendantType.ALL,
          auditContext,
          LockedInodePath::getInodeOrNull,
          (inodePath,  permChecker) -> permChecker.checkPermission(Mode.Bits.READ, inodePath),
          false);
```
 syncMetadata already avoid this case？

Reply:
change to O(n+m) time complexity.
Also I know syncMetadata will solves this problem, but syncMetadata relies on parameters configured on both the client and the server, and if it is set to never sync metadata as the default setting, then the inconsistency will always exist and we won't realize that the metadata is inconsistent. So I think the checkConsistency command should handle this case.

Reply:
Are we able to avoid this check case if syncMetada already synced and add this check case if not

Reply:
Add logic to determine if metadata is synchronized and skip the check.